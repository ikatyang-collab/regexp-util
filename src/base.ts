export abstract class Base {
  public isEmpty() {
    return this._isEmpty()
  }
  public toString() {
    if (this.isEmpty()) {
      throw new Error(`Output is empty.`)
    }
    return this._toString()
  }
  public toRegExp(flags?: string) {
    return new RegExp(this.toString(), flags)
  }

  protected abstract _isEmpty(): boolean
  protected abstract _toString(): string
}
