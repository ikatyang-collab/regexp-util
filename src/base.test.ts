import { test, expect } from 'vitest'
import { Base } from './base.js'

class Test extends Base {
  constructor(public data: string) {
    super()
  }
  protected _is_empty() {
    return this.data.length === 0
  }
  protected _to_string() {
    return this.data
  }
}

test('toRegExp: default', () => {
  expect(new Test('123').toRegExp()).toMatchSnapshot()
})

test('toRegExp: accept flags', () => {
  expect(new Test('123').toRegExp('i')).toMatchSnapshot()
})

test('toString: throw error if empty', () => {
  expect(() => new Test('').toString()).toThrowErrorMatchingSnapshot()
})
