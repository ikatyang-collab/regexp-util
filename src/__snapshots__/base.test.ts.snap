// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toRegExp: accept flags 1`] = `/123/i`;

exports[`toRegExp: default 1`] = `/123/`;

exports[`toString: throw error if empty 1`] = `"Output is empty."`;
